src/sagemaker/jumpstart/curated_hub/interfaces.py

```diff
@@ -0,0 +1,826 @@
+        self._region = region
+        self.dependencies = dependencies
+        self.from_json(json_obj)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Sets fields in object based on json.
+
+        Args:
+            json_obj (Dict[str, Any]): Dictionary representation of hub model document.
+        """
+        self.url: str = json_obj["Url"]
+        self.min_sdk_version: str = json_obj["MinSdkVersion"]
+        # Only one of ecr_uri and ecr_specs should exist.
+        self.hosting_ecr_uri: Optional[str] = json_obj["HostingEcrUri"]
+        self.hosting_artifact_uri = json_obj["HostingArtifactUri"]
+        self.hosting_script_uri = json_obj["HostingScriptUri"]
+        self.inference_dependencies: List[str] = json_obj["InferenceDependencies"]
+        self.inference_environment_variables: List[JumpStartEnvironmentVariable] = [
+            JumpStartEnvironmentVariable(env_variable, is_hub_content=True)
+            for env_variable in json_obj["InferenceEnvironmentVariables"]
+        ]
+        self.training_supported: bool = bool(json_obj["TrainingSupported"])
+        self.incremental_training_supported: bool = bool(json_obj["IncrementalTrainingSupported"])
+        self.dynamic_container_deployment_supported: Optional[bool] = (
+            bool(json_obj.get("DynamicContainerDeploymentSupported"))
+            if json_obj.get("DynamicContainerDeploymentSupported")
+            else None
+        )
+        self.hosting_artifact_s3_data_type: Optional[str] = json_obj.get(
+            "HostingArtifactS3DataType"
+        )
+        self.hosting_artifact_compression_type: Optional[str] = json_obj.get(
+            "HostingArtifactCompressionType"
+        )
+        self.hosting_prepacked_artifact_uri: Optional[str] = json_obj.get(
+            "HostingPrepackedArtifactUri"
+        )
+        self.hosting_prepacked_artifact_version: Optional[str] = json_obj.get(
+            "HostingPrepackedArtifactVersion"
+        )
+        self.hosting_use_script_uri: Optional[bool] = (
+            bool(json_obj.get("HostingUseScriptUri"))
+            if json_obj.get("HostingUseScriptUri") is not None
+            else None
+        )
+        self.hosting_eula_uri: Optional[str] = json_obj.get("HostingEulaUri")
+        self.hosting_model_package_arn: Optional[str] = json_obj.get("HostingModelPackageArn")
+        self.default_inference_instance_type: Optional[str] = json_obj.get(
+            "DefaultInferenceInstanceType"
+        )
+        self.supported_inference_instance_types: Optional[str] = json_obj.get(
+            "SupportedInferenceInstanceTypes"
+        )
+        self.sage_maker_sdk_predictor_specifications: Optional[JumpStartPredictorSpecs] = (
+            JumpStartPredictorSpecs(
+                json_obj.get("SageMakerSdkPredictorSpecifications"),
+                is_hub_content=True,
+            )
+            if json_obj.get("SageMakerSdkPredictorSpecifications")
+            else None
```

Comment:
Same q for all the values we use this pattern for: `Y if X else None`

Reply:
Same reply.